src/accelerate/accelerator.py

```diff
@@ -393,12 +396,11 @@ def __init__(
 
         self.delayed_fp8_autocast = False
         if self.fp8_recipe_handler is not None:
-            # We already check if FP8 is available during `self.state`
-            if self.state.mixed_precision != "fp8":
+            if self.state.mixed_precision != "fp8" and self.distributed_type != DistributedType.FSDP:
```

Comment:
We should maybe update the error message since you added an new condition ? Or we can do this new right below with its own error message.